src/createStore.ts

```diff
@@ -8,7 +12,36 @@ let createStoreAction = action('createStore', function createStoreAction(
     getRootStore().set(key, initialState);
 });
 
-export default function createStore<T>(key: string, initialState: T): () => T {
+export default function createStore<TState>(
+    key: string,
+    arg2: TState | LeafMutator<TState> | CombinedMutator<TState>
+): () => TState {
+    // Get the initial state (from the mutator, if necessary)
+    let mutator = getMutator(arg2);
+    let initialState = mutator ? mutator.getInitialValue() : arg2;
+
+    // Create the store under the root store
     createStoreAction(key, initialState);
-    return () => <T>getRootStore().get(key);
+    let getStore = () => <TState>getRootStore().get(key);
+
+    // If necessary, hook the mutator up to the dispatcher
+    if (mutator) {
+        subscribeAll(
```

Comment:
This seems really non-performant, if every mutator created in this new API is listening for every action.

Reply:
Given that this is all just pure JS and that for mutators that don't care about the action it's basically a NOP, I'm not too worried.  If it *does* turn out to be a problem I could imagine a way to have the mutators expose what actions they care about so we could subscribe to them individually...but I'm not going to add that complexity unless we know it's a problem.

---
In reply to: [200791963](https://github.com/Microsoft/satcheljs/pull/100#discussion_r200791963) [](ancestors = 200791963)